packages/melody-hooks/src/hooks/useState.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright 2018 trivago N.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { HOOK_TYPE_USE_STATE } from '../constants';
+import { getCurrentComponent, enterHook } from '../util/hooks';
+
+export const useState = initialState => {
+    enterHook();
+    const currentComponent = getCurrentComponent();
+    const { hooksPointer, hooks, state, setState } = currentComponent;
+
+    if (!currentComponent.isMounted) {
+        const setter = value => setState(hooksPointer, value);
```

Comment:
The setter should also support receiving a function which will do the update.
We should probably strive to be as compatible with React as possible to minimise confusion.

Reply:
done